sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java

```diff
@@ -279,99 +321,432 @@ public EventData addContext(String key, Object value) {
+     *
+     * @param correlationId correlation id of this message
+     *
+     * @return The updated {@link EventData}.
+     * @see #getCorrelationId()
+     */
+    public EventData setCorrelationId(String correlationId) {
+        AmqpMessageId id = null;
+        if (correlationId != null) {
+            id = new AmqpMessageId(correlationId);
+        }
+        amqpAnnotatedMessage.getProperties().setCorrelationId(id);
+        return this;
+    }
+
+    /**
+     * Gets the message id.
+     *
+     * <p>
+     * The message identifier is an application-defined value that uniquely identifies the message and its payload. The
+     * identifier is a free-form string and can reflect a GUID or an identifier derived from the application context.
+     * </p>
+     *
+     * @return Id of the {@link EventData}.
+     */
+    public byte[] getUserId() {
+        return amqpAnnotatedMessage.getProperties().getUserId();
+    }
+
+    /**
+     * Sets the message id.
+     *
+     * @param userId The message id to be set.
+     *
+     * @return The updated {@link EventData}.
+     * @throws IllegalArgumentException if {@code messageId} is too long.
+     */
+    public EventData setUserId(byte[] userId) {
+        amqpAnnotatedMessage.getProperties().setUserId(userId);
+        return this;
+    }
+
+    /**
+     * Gets the message id.
+     *
+     * <p>
+     * The message identifier is an application-defined value that uniquely identifies the message and its payload. The
+     * identifier is a free-form string and can reflect a GUID or an identifier derived from the application context.
+     * </p>
+     *
+     * @return Id of the {@link EventData}.
+     */
+    public String getMessageId() {
+        String messageId = null;
+        AmqpMessageId amqpMessageId = amqpAnnotatedMessage.getProperties().getMessageId();
+        if (amqpMessageId != null) {
+            messageId = amqpMessageId.toString();
+        }
+        return messageId;
+    }
+
+    /**
+     * Sets the message id.
+     *
+     * @param messageId The message id to be set.
+     *
+     * @return The updated {@link EventData}.
+     * @throws IllegalArgumentException if {@code messageId} is too long.
+     */
+    public EventData setMessageId(String messageId) {
+        checkIdLength("messageId", messageId, MAX_MESSAGE_ID_LENGTH);
+        AmqpMessageId id = null;
+        if (messageId != null) {
+            id = new AmqpMessageId(messageId);
+        }
+        amqpAnnotatedMessage.getProperties().setMessageId(id);
+        return this;
+    }
+
+    /**
+     * Gets the subject for the message.
+     *
+     * <p>
+     * This property enables the application to indicate the purpose of the message to the receiver in a standardized
+     * fashion, similar to an email subject line. The mapped AMQP property is "subject".
+     * </p>
+     *
+     * @return The subject for the message.
+     */
+    public String getSubject() {
```

Comment:
This should not be exposed as a top-level property.   The only additional properties that we should be adding are:

- Message Id
- Correlation Id
- Content Type

The other properties are available only via the AMQP message directly.    Please see [this issue](https://github.com/Azure/azure-sdk-for-net/issues/20105) for context.

Reply:
According to your comment, remove unnecessary parameters, variables and methods in the new version